graph/pull.go

```diff
@@ -88,6 +88,10 @@ func (s *TagStore) Pull(image string, tag string, imagePullConfig *ImagePullConf
 		logrus.Debug("image does not exist on v2 registry, falling back to v1")
 	}
 
+	if utils.DigestReference(tag) {
```

Comment:
While `utils.DigestReference` is a fragile function, it is strong enough in this case. It is only intended to test for digest given a reference (digest or tag). Reference should be made a stronger type with explicit digest or ref, but that is a different refactor which is not necessary to implement this fix.